### ISSUE TITLE
Shiny html representation for xgcm [FEATURE]

### ISSUE THREAD

**Issue by jbusecke** [xgcm/xgcm#276]
I think that it would be very beneficial to have a more visual representation of the grid object, similar to the HTML representation of xarray datasets/dataarrays.

I think this would help illustrate the rather abstract concepts of `axes`, `metrics` etc in a more user-friendly way. 

I drafted up a sketch of what this could look like for a 3 axis grid object (the most common case I think):
![My document (1)](https://user-images.githubusercontent.com/14314623/96152991-2701fb00-0edb-11eb-8c76-f8fd9fd5f4dc.jpg)
(This isnt my best work ahem, my Illustrator skills are a bit rusty these days...)

The central representation should be this wireframe cube (or rectangle, line for lower dimensions) which has the grid position highlighted. I am thinking we could then show additional information, when clicking/hovering over each grid point.

In the sketch I am showing for instance the metrics available for this specific position. We would show the name of each metric and if its available `natively` (e.g. provided by input) or `inferred` (e.g. formed as the product of lower level metrics or potentially inferred via interpolation of adjacent grid positions) or missing.
What I like about the 'wheel' representation is that you get a very good impression of which metrics are reconstructed, by looking at the intersection (e.g. `Area 2` is constructed by multipying `Distance 2` and `Distance 3`).

In that pop up box we could add more info like: 
Which data variables are situated on this point?
...there are probably other things that we could put in there too

Id love to get some feedback on the initial idea. This will be a longer term project, but I thought Ill start the discussion. This was in part motivated by our planned overhaul of the metrics internals in #199 (@dcherian I hope some of my suggested changes make more sense with this in mind).

**Comment by rabernat** [comment]
Really great idea! This could leverage some of dask's repr_html SVG machinery to draw the cubes.

**Comment by jbusecke** [comment]
Yes, that would be awesome. I am not very familiar with this, but will try to read into this. If anyone has experience in these sort of things, Id really appreciate some help in building this.

**Comment by dcherian** [comment]
Ah this would've been a great intern project for someone with HTML/SVG/CSS skills

**Comment by rabernat** [comment]
"would've"?

**Comment by jbusecke** [comment]
That is a great suggestion in general though @dcherian. Lets keep our eyes peeled for upcoming opportunities. This does not need to happen immediately, but is a nice and constrained project in and by itself.

**Comment by miniufo** [comment]
I love the 'pop up' feature!!!  Visualization is always great.
